### PR TITLE
Feature/react settings pages

### PR DESCRIPTION
**Pull request by vedrani**

### CODE REVIEW

**Comment by vedrani** [comment]
This is just 1st part...

File: docs/extensions/tutorials/_posts/1970-01-051-DebugSettingsPages.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+If you try to develop [settings pages]({{ site.url }}/docs/extensions/tutorials/writing-settings-page) you will know how to create extension, add settings page, reference it, push it so it's visible in Shoutem Builder. After that every change in code is only visible in Builder if you push changes with `shoutem push` and Builder reload in browser.
+
+To make your development experience efficient and comfortable as much as possible we introduced `Debug` button on each settings page in Builder that you own as developer.
+
+![Developer settings page with Debug button]({{ site.url }}/img/tutorials/writting-settings-page/hello-world-settings-page.png "Developer settings page with Debug button"){:.docs-component-image}
+
+Now, when you develop settings pages all you need to do is few short steps:
+
+1. Push, and install your extension in Shoutem Builder application (Only once)
+2. In terminal in extension directory, execute `shoutem run server-dev` to bundle your extension and run webpack dev server (Until you kill server there is no need to run it again)
```
Comment (SoHotSoup) [inline docs/extensions/tutorials/_posts/1970-01-051-DebugSettingsPages.md]: @vedrani `shoutem run server-dev` doesn't work at the moment. should be announce it and add note that you should execute `npm run dev`?

File: docs/extensions/tutorials/_posts/1970-01-051-DebugSettingsPages.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+3. In your browser go to Shoutem Builder and open settings page you are developing
+4. Click on Debug button
+
+Once you click Debug, settings page will reload from `https://localhost:4790` and there is possibility that browser will define `localhost` as insecure. Open `https://localhost:4790/index.html` in browser and add exception for insecure page or in Chrome open `chrome://flags/#allow-insecure-localhost.` and enable it.
```
Comment (SoHotSoup) [inline docs/extensions/tutorials/_posts/1970-01-051-DebugSettingsPages.md]: I would add a note that sometimes is needed to restart chrome after enabling allow insecure localhost flag. Also navigating to link lead me to the blank page, and I had to navigate to`chrome://flags`

File: docs/extensions/tutorials/_posts/1970-01-051-DebugSettingsPages.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+As we use React and Webpack dev server we have hot module reload feature enabled. What's cool about it is that every time you change your code in editor and save it, server will rebundle code and reload it browser, so basicallz you have a live preview while you develop your settings page. There are some exceptions when it cannot work, like adding new npm module, file, but in such case just re-run dev server with `shoutem run server-dev` and reload Builder in browser.
```
Comment (SoHotSoup) [inline docs/extensions/tutorials/_posts/1970-01-051-DebugSettingsPages.md]: `npm run dev`

File: docs/extensions/tutorials/_posts/1970-01-10-DebugSettingsPages.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+<hr />
+
+This short tutorial covers how to improve your settings page development flow by utilizing the `Debug` feature in the Builder. It'll explain how to see code changes without having to `push` every code change and how to use dev tools to debug the settings pages through the Builder.
+
+> #### Note
+> HTML settings pages currently cannot use the `Debug` feature in the Builder.
+
+## Local development
+
+When you start developing settings pages, you'll quickly realize how much of a hassle it is to `shoutem publish` every change to see the updates on the Builder. To avoid this and enjoy a much more efficient development flow, you can simply do the following:
```
Comment (dodsky) [inline docs/extensions/tutorials/_posts/1970-01-10-DebugSettingsPages.md]: When you start developing settings pages, you'll quickly realize that it's inpractical to `shoutem publish` every change to Shoutem just to see the updated settings page.

File: docs/extensions/tutorials/_posts/1970-01-10-DebugSettingsPages.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+1. Push, and install your extension in Shoutem Builder application (Only once)
+2. Locate to the `extName/server` directory and execute `npm run dev` to run the webpack dev server (Until you kill server there is no need to run it again)
```
Comment (dodsky) [inline docs/extensions/tutorials/_posts/1970-01-10-DebugSettingsPages.md]: This is too much to remember. This is how I'd like it to work:
1. click debug icon in builder
A modal is shown instructing you how to serve your settings page locally by instructing you to execute a command in shoutem CLI positioned inside of extension directory. The command should display something like: `shoutem page debug <page-name>`, CLI would then run `npm run dev` inside of the appropriate server folder. Command should be copy/pasteable with copy to clipboard button. It should also display a link to this page, where security setup is explained.

2. paste the command in shoutem CLI positioned at extension folder, it runs dev env

3. develop :)

Comment (dodsky) [reply docs/extensions/tutorials/_posts/1970-01-10-DebugSettingsPages.md]: @vedrani @ikatun what do you think? 🔝

**Pull request merged**